### PR TITLE
Mock NextAuth.js and fix issues in session about NYCU auth

### PR DESCRIPTION
**Pull request by alan910127**
The solution to mocking next auth is not perfect, but I think it's acceptable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,3 +14,9 @@ Score management made easy
 ## Contribution
 
 Commit message: prefix with an [meaningful emoji](https://gitmoji.dev)
+
+## Mocking Next Auth during Development
+
+To mock NextAuth in development, you should set the environment variable `NEXT_PUBLIC_MOCK_ROLE` to `admin`, `ta` or `student` in your `.env` file.
+
+⚠️ Switching between mocking or not needs manual refresh on the client side. ⚠️
```
Comment (alan910127) [inline README.md:22]: The needs of refreshing manually is caused by caching from `@tanstack/react-query` (used by `trpc`).

File: src/providers/nycu.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export default function NYCUProvider<P extends NYCUProfile>({
     profile(profile) {
       return {
         id: profile.username,
+        name: profile.username,
```
Comment (alan910127) [inline src/providers/nycu.ts:28]: The `username` returned by NYCU auth is actually student id

**Review by alan910127 — COMMENTED** [review]
Sorry for so many changes

File: README.md (hunk 1 above)
Comment (BaBa0525) [reply README.md:22]: good

File: src/providers/nycu.ts (hunk 2 above)
Comment (BaBa0525) [reply src/providers/nycu.ts:28]: good

**Review by BaBa0525 — APPROVED** [review]
Good Job!

**Pull request merged**